sensor/common/processsignal/pipeline.go

```diff
@@ -142,16 +154,41 @@ func (p *Pipeline) sendIndicatorEvent() {
 			continue
 		}
 		p.detector.ProcessIndicator(p.getCurrentContext(), indicator)
-		p.indicators <- message.NewExpiring(p.getCurrentContext(), &central.MsgFromSensor{
-			Msg: &central.MsgFromSensor_Event{
-				Event: &central.SensorEvent{
-					Id:     indicator.GetId(),
-					Action: central.ResourceAction_CREATE_RESOURCE,
-					Resource: &central.SensorEvent_ProcessIndicator{
-						ProcessIndicator: indicator,
+		p.sendToCentral(
+			message.NewExpiring(p.getCurrentContext(), &central.MsgFromSensor{
```

Comment:
nit: it's not a big deal, but technically you don't need a lock when the FF is enabled. But `p.getCurrentContext` will use a write mutex just to get a pointer to `context.Background()`.

Reply:
Good call. I've changed `getCurrentContext` to return `Background` if we are in v3.